lib/request-wrapper.ts

```diff
@@ -311,6 +320,29 @@ export class RequestWrapper {
 
     return error;
   }
+
+  private gzipRequestBody(data: Object, headers: OutgoingHttpHeaders): Buffer|Object {
```

Comment:
If gzip compression is enabled, we should compress the request body regardless of the type of data that is present.  i.e. this would apply to more than just JSON objects that need to be stringified, such as a text/plain or application/octet stream type request body.